### PR TITLE
Hardware keyboard support (Windows support + Mac fix)

### PR DESCRIPTION
**Pull request by reneklacan**
Added Windows support for HW keyboard (as addition to https://github.com/cocos2d/cocos2d-x/pull/2704) and also included fix for Mac.

### CODE REVIEW

**Comment by dumganhar** [comment]
Could you pull the latest codes?

**Comment by reneklacan** [comment]
sure, done :)

File: cocos2dx/platform/win32/CCEGLView.cpp [hunk 1]
```diff
@@ -449,6 +452,9 @@ LRESULT EGLView::WindowProc(UINT message, WPARAM wParam, LPARAM lParam)
         }
         break;
     case WM_KEYDOWN:
+#ifdef KEYBOARD_SUPPORT
```
Comment (dumganhar) [inline cocos2dx/platform/win32/CCEGLView.cpp]: We have renamed `KEYBOARD_SUPPORT` to `CC_ KEYBOARD_SUPPORT`.
Do you mind to change it to `CC_ KEYBOARD_SUPPORT`?

Comment (reneklacan) [inline cocos2dx/platform/win32/CCEGLView.cpp]: no problem, I will change that :)

Comment (reneklacan) [inline cocos2dx/platform/win32/CCEGLView.cpp]: done

**Comment by reneklacan** [comment]
also please consider merge of https://github.com/cocos2d/cocos2d-x/pull/2776

**Comment by minggo** [comment]
Thanks @simpliplant.
I have asked @boyu0 to review #2776.

**Comment by rishftw** [comment]
okay guys, I'm really a big noob, so sorry, but how do i use only the cross-platform keyboard feature, without using the develop brach? like, can i just add the keyboard part to cocos2d-x 2.1.4?
I don't wanna use the develop branch cuz I'm guessing there are problems?:/
thanks

**Pull request merged**